lesson_26/api/java/api_app/src/main/java/com/codedifferently/lesson26/web/MediaItemsController.java

```diff
@@ -29,4 +34,31 @@ public GetMediaItemsResponse getItems() {
     var response = GetMediaItemsResponse.builder().items(responseItems).build();
     return response;
   }
+
+  @PostMapping("/items")
+  public MediaItemResponse addItemById(@RequestBody CreateMediaItemRequest createRequest) {
+    if (createRequest.getItem().getId() == null) {
+      createRequest.getItem().setId(UUID.randomUUID());
+    }
+    
+      library.add(createRequest.getItem());
+    
+    return  MediaItemResponse.from(createRequest.getItem());
+  }
+
+  @GetMapping("/items/:{id}")
+  public MediaItemResponse getItemById(@PathVariable("id") UUID id) {
+    MediaItem item = library.search(SearchCriteria.builder().id(id.toString()).build()).stream().findFirst().orElseThrow();
```

Comment:
You don't want the `.orElseThrow()` part because that won't allow you to send the correct response. Instead, you should return a `404 NOT FOUND` response. You can accomplish this by returning a `ResponseEntity` from this function instead of `MediaItemResponse` directly.